src/utilities/theme-utils.ts

```diff
@@ -22,7 +22,21 @@ export const setThemeColors = (primary: string, secondary: string) => {
   document.documentElement.style.setProperty("--theme-primary-color", primary);
   document.documentElement.style.setProperty("--theme-primary-hover-color", colorShade(primary, -20));
   document.documentElement.style.setProperty("--theme-primary-active-color", colorShade(primary, -40));
+  document.documentElement.style.setProperty("--theme-primary-text-color", getContrastYIQColor(primary));
   document.documentElement.style.setProperty("--theme-secondary-color", secondary);
   document.documentElement.style.setProperty("--theme-secondary-hover-color", colorShade(secondary, -20));
   document.documentElement.style.setProperty("--theme-secondary-active-color", colorShade(secondary, -40));
+  document.documentElement.style.setProperty("--theme-secondary-text-color", getContrastYIQColor(secondary));
+};
+
+// example from:
+// https://24ways.org/2010/calculating-color-contrast
+const getContrastYIQColor = (color: string) => {
+  color = color.replace(/^#/, "");
+  if (color.length === 3) color = color[0] + color[0] + color[1] + color[1] + color[2] + color[2];
+  const r = parseInt(color.substr(0,2),16);
+  const g = parseInt(color.substr(2,2),16);
+  const b = parseInt(color.substr(4,2),16);
+  const yiq = ((r * 299) + (g * 587) + (b * 114)) / 1000;
+  return (yiq >= 128) ? "#222222" : "#eeeeee";
```

Comment:
I'd be inclined to use named constants for these rather than magic numbers. Ideally, they'd be shared with the SCSS, but I'd settle for named constants in the short term.

Reply:
Good call.  I added them to the .scss file and imported them from there using the same method you came across for use in CLUE (and I think Chris used in CBio).